app/main.py

```diff
@@ -1 +1,96 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (NotVaccinatedError, OutdatedVaccineError,
+                        NotWearingMaskError)
+import datetime
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_needed = 0
+    all_vaccinated = True
+
+    for friend in friends:
+        try:
+            if "vaccine" not in friend:
+                all_vaccinated = False
+                continue
```

Comment:
Instead of checking for the presence of 'vaccine' and manually setting `all_vaccinated` to `False`, you should raise a `NotVaccinatedError` here. This will allow the `try/except` block to handle the error as intended.